### PR TITLE
fix: Correct STRICT_ENV_VAR_MODE behavior

### LINKED ISSUE #6604: Bug: Env var strict mode not working as expected
While [testing](https://github.com/meltano/squared/pull/312#issuecomment-1191828551) within the `squared` project, and also confirmed by @visch [in Slack](https://meltano.slack.com/archives/CFG3C3C66/p1660078390407119?thread_ts=1659968745.750009&cid=CFG3C3C66), we've not been seeing the strict mode flag working as expected to raise errors.

It could be the case, in the multiple recent changes to this part of the codebase (`env` resolution, specifically) that we may have had a regression.

### PR DESCRIPTION
**Pull request by cjohnhanson**
The `strict_env_var_mode` feature flag hasn't been behaving correctly since the 2.3.0 release. This 
PR fixes the behavior and adds new functional tests in addition to the existing unit tests to catch 
such regressions moving 
forward.

Closes #6604

### CODE REVIEW

File: src/meltano/core/plugin_invoker.py [hunk 1]
```diff
@@ -327,7 +327,7 @@ def env(self):
         project_settings_service = ProjectSettingsService(
             self.project, config_service=self.plugins_service.config_service
         )
-        with self.settings_service.feature_flag(
+        with project_settings_service.feature_flag(
```
Comment (edgarrmondragon) [inline src/meltano/core/plugin_invoker.py:330]: I think this might fix #6485 too 🤞

**Review by edgarrmondragon — APPROVED** [review]
LGTM!

**Review by pandemicsyn — APPROVED** [review]
🚀

File: src/meltano/core/plugin_invoker.py (hunk 1 above)
Comment (cjohnhanson) [reply src/meltano/core/plugin_invoker.py:330]: 🤞

**Pull request merged**